syntax = "proto3";

package keyvalue;

service KeyValueService {
  rpc Insert (KeyValueRequest) returns (KeyValueResponse);
  rpc Consult (KeyRequest) returns (ValueResponse);
  rpc Activate (ServiceRequest) returns (EmptyResponse);
  rpc Terminate (EmptyRequest) returns (EmptyResponse);  
}

message KeyValueRequest {
  int key = 1;
  string value = 2;
}

message KeyValueResponse {
  int result = 1;
}

message KeyRequest {
  int key = 1;
}

message ValueResponse {
  string value = 1;
}

message EmptyRequest {}

message EmptyResponse {}
